### PR TITLE
Add support for appended zip files

### PR DESCRIPTION
**Pull request by liron-shalom**

### CODE REVIEW

File: archive_extractor/zip_archiver.go [hunk 1]
```diff
@@ -52,3 +61,65 @@ func (za ZipArchiver) ExtractArchive(path string,
...
+// it searches for begin-of-file signature 0x04034b50 (in little-endian the value of the signature is PK0304).
+// Once the signature was found, tries to read the file starting at that offset using zip.NewReader.
+func initZipReader(r io.ReaderAt, size int64) (*zip.Reader, error) {
+	zr, err := zip.NewReader(r, size)
+	if err == nil || !errors.Is(err, zip.ErrFormat) {
+		return zr, err
+	}
+	const BUFSIZE = 4096
+	var buf [BUFSIZE + 4]byte
+	for i := int64(0); (i-1)*BUFSIZE < size; i++ {
```
[the remark this answers is not in the document]
Comment (galusben) [reply archive_extractor/zip_archiver.go:101]: more tests, super small zip

[the remark this answers is not in the document]
Comment (galusben) [reply archive_extractor/zip_archiver.go:101]: empty zip

File: archive_extractor/zip_archiver.go [hunk 2]
```diff
@@ -52,3 +61,65 @@ func (za ZipArchiver) ExtractArchive(path string,
...
+		len, err := r.ReadAt(buf[:], i*BUFSIZE)
+		if err != nil && err != io.EOF {
+			break
+		}
+		n := 0
+		for {
```
Comment (galusben) [inline archive_extractor/zip_archiver.go:107]: I don't understand why this loop is necessary.

File: archive_extractor/zip_archiver.go [hunk 3]
```diff
@@ -52,3 +61,62 @@ func (za ZipArchiver) ExtractArchive(path string,
...
+// it searches for begin-of-file signature 0x04034b50 (in little-endian the value of the signature is PK0304).
+// Once the signature was found, tries to read the file starting at that offset using zip.NewReader.
+func initZipReader(r io.ReaderAt, size int64) (*zip.Reader, error) {
+	zr, err := zip.NewReader(r, size)
+	if err == nil || !errors.Is(err, zip.ErrFormat) {
+		return zr, err
+	}
+	const BUFSIZE = 4096
+	var buf [BUFSIZE + 4]byte
+	for i := int64(0); (i-1)*BUFSIZE < size; i++ {
```
Comment (asafambar) [inline archive_extractor/zip_archiver.go:98]: if size is equal to `BUFSIZE*1.5` , will I read all the zip file ?

**Review by galusben — APPROVED** [review]

**Pull request merged**